playwright_test/playwright.config.js

```diff
@@ -1,109 +1,42 @@
+// playwright.config.js
 // @ts-check
 const { devices } = require('@playwright/test');
 
-/**
- * Read environment variables from file.
- * https://github.com/motdotla/dotenv
- */
-// require('dotenv').config();
-
-
-/**
- * @see https://playwright.dev/docs/test-configuration
- * @type {import('@playwright/test').PlaywrightTestConfig}
- */
+/** @type {import('@playwright/test').PlaywrightTestConfig} */
 const config = {
-  testDir: './tests',
-  /* Maximum time one test can run for. */
-  timeout: 30 * 1000,
-  expect: {
-    /**
-     * Maximum time expect() should wait for the condition to be met.
-     * For example in `await expect(locator).toHaveText();`
-     */
-    timeout: 5000
-  },
-  /* Run tests in files in parallel */
-  fullyParallel: true,
-  /* Fail the build on CI if you accidentally left test.only in the source code. */
-  forbidOnly: !!process.env.CI,
-  /* Retry on CI only */
-  retries: process.env.CI ? 2 : 0,
-  /* Opt out of parallel tests on CI. */
-  workers: process.env.CI ? 1 : undefined,
-  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
-  reporter: 'html',
-  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  // Options shared for all projects.
+  timeout: 30000,
   use: {
-    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
-    actionTimeout: 0,
-    /* Base URL to use in actions like `await page.goto('/')`. */
-    // baseURL: 'http://localhost:3000',
-
-    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
-    trace: 'on-first-retry',
+    ignoreHTTPSErrors: true,
+    headless: false,
   },
 
-  /* Configure projects for major browsers */
+  // Options specific to each project.
   projects: [
     {
-      name: 'chromium',
+      name: 'Desktop Chromium',
       use: {
-        ...devices['Desktop Chrome'],
+        browserName: 'chromium',
+        viewport: { width: 1280, height: 720 },
+        permissions: ['geolocation'],
       },
     },
-
     {
-      name: 'firefox',
+      name: 'Desktop Safari',
       use: {
-        ...devices['Desktop Firefox'],
-      },
+        browserName: 'webkit',
+        viewport: { width: 1280, height: 720 },
+      }
```

Comment:
you haven't give any permission for geo locations

Reply:
not required for WebKit